torchinductor/codegen/triton.py

```diff
@@ -571,23 +604,11 @@ def var_ranges(self):
             ),
         )
 
-    def simplify_indexing(self, expr: sympy.Expr):
+    def codegen_indexing(self, expr: sympy.Expr):
         expr = V.graph.sizevars.simplify_with_ranges(expr, self.var_ranges())
-        sorted_expr_symbols = sorted(expr.free_symbols, key=lambda s: s.name)
-        nodes = [
-            self.range_tree_nodes[sym]
-            for sym in sorted_expr_symbols
-            if sym in self.range_tree_nodes
-        ]
-        if nodes:
-            nodes.sort(key=lambda x: x.depth)
-            expr = nodes[-1].simplify(expr)
-            sorted_expr_symbols = sorted(expr.free_symbols, key=lambda s: s.name)
-
-        for sym in sorted_expr_symbols:
+        for sym in expr.free_symbols:
```

Comment:
and here, expr.free_symbols is a set, so it will be iterated in arbitrary order

Reply:
fixed